core/src/main/java/org/infinispan/functional/impl/FunctionalMapFactory.java

```diff
@@ -0,0 +1,24 @@
+package org.infinispan.functional.impl;
+
+import org.infinispan.functional.FunctionalMap;
+
+/**
+ * Helper Factory to create FunctionalMaps
```

Comment:
Helper for whom, when it's in impl package? If it's for tests, put it under `test/`

Reply:
At first there was more code than what i is right now. But I prefer 
`FunctionalMapFactory.readOnlyMap(functionalMap)`
that telling the user to use `ReadOnlyMapImpl.create(functionalMap);`
Like having a single entry point. But well, this is my opinion, and this class can be removed

Reply:
I dislike `new ReadOnlyMapImpl(fmap)` as well - but there should be only one recommended way (for users, tests and internal code alike) to create the impl.